presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/RowExpressionVerifier.java

```diff
@@ -522,8 +522,11 @@ protected Boolean visitLogicalBinaryExpression(LogicalBinaryExpression expected,
             SpecialFormExpression actualLogicalBinary = (SpecialFormExpression) actual;
             if ((expected.getOperator() == OR && actualLogicalBinary.getForm() == SpecialFormExpression.Form.OR) ||
                     (expected.getOperator() == AND && actualLogicalBinary.getForm() == SpecialFormExpression.Form.AND)) {
-                return process(expected.getLeft(), actualLogicalBinary.getArguments().get(0)) &&
-                        process(expected.getRight(), actualLogicalBinary.getArguments().get(1));
+                // `Logical AND` and `Logical OR` both satisfy the commutative property
+                return process(expected.getLeft(), actualLogicalBinary.getArguments().get(0)) ?
```

Comment:
I just want to provide an example so that I understand this change.

Given 

```
actual: x < 10 AND y > 2

expected: y > 2 AND x < 10
```

Previously, this would have been false given that the arguments were flipped. However, with this change once the first check fails for the left expressions being not-equal, it attempts the check again, flipping the expressions checked.

This makes sense to me, but is there anything in the row expression library which canonicalizes expressions first such that the ordering would be consistent before being sent to the verifier? I'm not all that familiar with the `RowExpressions` part of the codebase, but I would be surprised if something to re-order these expressions didn't already exist.

The main reason I ask is because if it is the case that expressions can be canonicalized/sorted, is this change really necessary?

Reply:
In this PR, we try to split the original filter expression into several parts: identity partitioned columns expression, non-identity entire column expression, subfield expression, unable-to-extract expression. Then we pushed down and discard the part which could be enforced by iceberg table, and combine the rest parts to re-create a filter expression for FilterNode. The main reason of the re-ordered filter expression left in FilterNode is that the combination sequence is based on the category of each part rather than the sequence appears in original filter expression.

So, if we have a filter consist of expressions of different categories, we may have a re-ordered filter expression after the partial pushdown optimization logic. For example, if the original filter is `a = 10 and b = 11 and r.c = 12` where column `a` is identity partition column, after partial pushdown logic the left filter would be `r.c = 12 and b = 11`.
Because we combine the left filter expression using the following code:

```
RowExpression remainingFilterExpression = logicalRowExpressions.combineConjuncts(
                    ImmutableList.of(
                            decomposedFilter.getRemainingExpression(),
                            subfieldPredicate,
                            nonPartitionColumn));
```

I think it's useless to keep their original sequence of  appearance. What's your opinion?

Reply:
By the way, seems that previously the expressions would be send to the verifier un-canonicalized, that is, keeping their originally appearance order. So we would get a  row expression `((id) = (INTEGER'10')) AND ((name) = (VARCHAR'adam'))` for `id = 10 and name = 'adam'`, and get a row expression `(((name) = (VARCHAR'adam') AND (id) = (INTEGER'10')))` for `name = 'adam' and id = 10`, and they would be asserted not equal by the verifier.

Reply:
Thanks, this explanation helped clear up my understanding of the PR as a whole. If there is no canonicalization already available, then I think this change is good 👍